Source/RxComponents/LMCCavityModes.cc

```diff
@@ -61,15 +61,31 @@ namespace locust
             }
         }
 
+
+        fVoltagePhase.resize(2); // max nchannels.  TO-DO:  Configure this as LMCGenerator::fNChannels.
```

Comment:
It looks like the resizing doesn't line up with the indexing here in general. You start by resizing fVoltagePhase for 2 channels, but then iterate over nModes. Here they happen to both be 2, but I think the first for loop should index over nchannels, the second and third loops should index over GetNCavityModes(), and then you won't need a third GetNCavityModes() loop. Still 3 for loops total though.